Backend/src/main/java/com/example/backend/project/service/ProjectService.java

```diff
@@ -24,82 +29,137 @@ public class ProjectService {
     private UserRepository userRepository;
     private CollaborationsRepository collaborationsRepository;
 
-    public List<Project> findAll() {
+    public List<Project> findAll(AppUser user) throws AuthenticationException {
+        if (user == null) throw new AuthenticationException();
+
         return projectRepository.findAll();
     }
 
-    public Project findById(Long id) {
-        return projectRepository.findById(id).get();
+    public ProjectAndFRTeamMembersDPO findById(Long id, AppUser user) throws AuthenticationException, ProjectNotFoundException {
+        if (user == null) throw new AuthenticationException();
+
+        Project project;
+        if (projectRepository.findById(id).isPresent()) project = projectRepository.findById(id).get();
+        else throw new ProjectNotFoundException();
+        Set<FRTeamMemberDPO> frTeamMemberDPOS = new HashSet<>();
+        for (AppUser appUser: project.getFrteammembers()){
```

Comment:
Ovakve stvari je možda bolje napraviti pomoću https://www.oracle.com/technical-resources/articles/java/ma14-java-se-8-streams.html. U c# se to zove linq-u, pa nisam 100% uvjeren možeš či sve dohvatiti, mappirati, pa staviti u listu, ali čini mi se dovoljno slično da bi mogao. Repozzitory bi ti automatski trebao imati transaction, pa možeš samo ići projectRepository.findById(id).map(x => new FRTeamMemberDPO()) ili kako već sintaksa ide. Prednosti ovoga su urednost i brzina(brzina upitno, u c# je brže, jer veći dio toga radi u bazi, pretpostavljam da steam radi isto), Predlažem da malo pokušaš to tako implementirati, ali ako ti iz prve ne ide ignoriraj.

Reply:
zanimljivo, probam i ovo

Reply:
Ovo me zanima hoćeš li uspjeti jednostavno koristiti. Ako da, molim te da mi javiš. Ako ne pokušati ću to malo više proučiti kada se uhodam malo u projekt.